conans/tools.py

```diff
@@ -165,6 +165,17 @@ def build_sln_command(settings, sln_path, targets=None, upgrade_project=True, bu
         command += " /target:%s" % ";".join(targets)
     return command
 
+def vs_installation_path(version):
+    if not hasattr(vs_installation_path, "_cached"):
```

Comment:
What exception will raise in case it fails? Will fail the subprocess? will fail to access to the dict? if the exception raised is not a ConanException I would prefer to keep it under control and return None, but please do not "except Exception".